apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -506,6 +511,12 @@ type MetricSpec struct {
 	Pods *autoscalingv2.PodsMetricSource `json:"pods,omitempty"`
 }
 
+type ConfigMapsSpec struct {
+	// Configmap defines name and path where the configMaps should be mounted.
+	Name      string `json:"name,omitempty"`
+	MountPath string `json:"mountpath,omitempty"`
```

Comment:
Both of these should be required and nonempty. Looks like MountPath should be at least `/`, and I'd use some minimum character count for name as well.

Reply:
Regarding the name's length, I've used the `DNSName` function to validate its length.